packages/eas-cli/src/credentials/manager/SelectIosDistributionTypeGraphqlFromIosDistributionTypeEasConfig.ts

```diff
@@ -0,0 +1,65 @@
+import { IosDistributionType as IosDistributionTypeEasConfig } from '@expo/eas-json';
+import _ from 'lodash';
+
+import {
+  CommonIosAppCredentialsFragment,
+  IosDistributionType as IosDistributionTypeGraphql,
+} from '../../graphql/generated';
+import { promptAsync } from '../../prompts';
+import { Context } from '../context';
+
+export class SelectIosDistributionTypeGraphqlFromIosDistributionTypeEasConfig {
+  constructor(private iosDistributionTypeEasConfig: IosDistributionTypeEasConfig) {}
+  async runAsync(
+    ctx: Context,
+    iosAppCredentials: CommonIosAppCredentialsFragment | null
+  ): Promise<IosDistributionTypeGraphql> {
+    if (this.iosDistributionTypeEasConfig === 'store') {
+      return IosDistributionTypeGraphql.AppStore;
+    } else if (this.iosDistributionTypeEasConfig === 'simulator') {
+      return IosDistributionTypeGraphql.Development;
+    }
+    return this.resolveInternalDistributionAsync(ctx, iosAppCredentials);
+  }
+
+  async resolveInternalDistributionAsync(
+    ctx: Context,
+    iosAppCredentials: CommonIosAppCredentialsFragment | null
+  ): Promise<IosDistributionTypeGraphql> {
+    const isDefinitelyNotAnEnterpriseAccount = ctx.appStore.authCtx?.team.inHouse === false;
+    if (isDefinitelyNotAnEnterpriseAccount) {
+      return IosDistributionTypeGraphql.AdHoc;
+    }
+    const iosAppBuildCredentialsArray = iosAppCredentials?.iosAppBuildCredentialsArray ?? [];
+    const existingInternalIosDistributionTypes = _.uniq(
+      iosAppBuildCredentialsArray
+        .map(buildCredentials => buildCredentials.iosDistributionType)
+        .filter(
+          distributionTypeGraphql =>
+            distributionTypeGraphql === IosDistributionTypeGraphql.AdHoc ||
+            distributionTypeGraphql === IosDistributionTypeGraphql.Enterprise
+        )
+    );
+    if (existingInternalIosDistributionTypes.length === 1) {
+      return existingInternalIosDistributionTypes[0];
+    }
+
```

Comment:
You can check the `universalDistribution` property from eas.json before displaying the prompt below.
https://github.com/expo/eas-cli/pull/335 added `"universalDistribution": "adhoc" | "universal"` to eas.json which is handling this case. I'd only display the prompt if `universalDistribution` is not specified.

Reply:
did u mean `enterpriseProvisioning`?

Reply:
@quinlanj yes 😅